ci-operator/step-registry/cucushift/installer/check/azure/identity/cucushift-installer-check-azure-identity-commands.sh

```diff
@@ -112,59 +112,97 @@ check_result=0
-    for num in $(seq 0 ${end_number}); do
-        subscrption=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].subscription")
-        name=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].name")
-        rg=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].resourceGroup")
-        compute_identity_id+=" /subscriptions/${subscrption}/resourceGroups/${rg}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${name}"
-    done
+
+if [[ -n "${identity_type_worker}" ]]; then
+    expected_identity_type_worker="${identity_type_worker}"
+    if [[ "${identity_type_worker}" == "UserAssigned" ]] && [[ ${AZURE_USER_ASSIGNED_IDENTITY_NUMBER} -gt 0 ]]; then
+        expected_identity_id_worker=""
+        for num in $(seq 0 ${end_number}); do
+            subscrption=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].subscription")
+            name=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].name")
+            rg=$(yq-go r ${INSTALL_CONFIG} "compute[0].platform.azure.identity.userAssignedIdentities[$num].resourceGroup")
+            expected_identity_id_worker+=" /subscriptions/${subscrption}/resourceGroups/${rg}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${name}"
+        done
+    fi
 fi
 
-echo "control_plane_identity_id: ${control_plane_identity_id}"
-echo "compute_identity_id: ${compute_identity_id}"
+echo "expected_identity_type_master: ${expected_identity_type_master}"
+echo "expected_identity_id_master: ${expected_identity_id_master}"
+echo "expected_identity_type_worker: ${expected_identity_type_worker}"
+echo "expected_identity_id_worker: ${expected_identity_id_worker}"
+echo "cluster_identity_id: ${cluster_identity_id}"
+echo "AZURE_USER_ASSIGNED_IDENTITY_NUMBER: ${AZURE_USER_ASSIGNED_IDENTITY_NUMBER}"
+
+if [[ "${expected_identity_type_master}" == "UserAssigned" ]] && [[ -z "${expected_identity_id_master}" ]]; then
+    echo "control plane's identity type is UserAssigned, but expected identity ids are empty, please check..."
+    exit 1
+fi
+
+if [[ "${expected_identity_type_worker}" == "UserAssigned" ]] && [[ -z "${expected_identity_id_worker}" ]]; then
+    echo "compute's identity type is UserAssigned, but expected identity ids are empty, please check..."
+    exit 1
+fi
 
-if [[ -z "${control_plane_identity_id}" ]] && [[ -z "${compute_identity_id}" ]] && (( ${ocp_minor_version} >= 19 )); then
-    echo "No identity is specified in install-config and no user-assigned identity is created by default on 4.19+, skip the check..."
-    exit 0
+if [[ -z "${expected_identity_type_master}" ]] && [[ -z "${expected_identity_type_worker}" ]] && [[ -z "${cluster_identity_id}" ]]; then
+    echo "No identity is specified, installer should create an identity by default, but not found it, exit..."
```

Comment:
I guess it is allowed to set `controlPlane.platform.azure.identity.type=UserAssigned` but without any customer identity, right ?